blockchain/hash_test.go

```diff
@@ -0,0 +1,160 @@
+package blockchain
+
+import (
+	"testing"
+)
+
+func TestParseString(t *testing.T) {
+	var h Hash
+	if h.ParseString("") != nil {
+		t.Fail()
+	}
+
+	if h.ParseString("1") != nil {
+		t.Fail()
+	}
+
+	if h.ParseString("a") != nil {
+		t.Fail()
+	}
+
+	if h.ParseString("ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff") != nil {
```

Comment:
@david-julien Is it possible to replace these huge hex strings with `math.big`. I don't like putting big string literal hexes all over the code.

Rather than doing `hash.ParseString()` to get a hash from a big number, can we use the `math.big` exponentiation function like `z.Exp(2, 50, 0)` to get 2^50.

If we need the ParseString function, do `func ParseString(s string) Hash` rather than having the hash as the receiver.